bonnie-rest/src/main/java/com/cgi/bonnie/bonnierest/UserRegister.java

```diff
@@ -0,0 +1,29 @@
+package com.cgi.bonnie.bonnierest;
+
+import com.cgi.bonnie.bonnierest.model.UserRequest;
+import com.cgi.bonnie.businessrules.Role;
+import com.cgi.bonnie.businessrules.user.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/user")
+public class UserRegister {
+
+    final private UserService userService;
+
+    @Autowired
+    public UserRegister(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping(path = "/register", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public ResponseEntity<String> createUser(UserRequest request) {
+        long id = userService.createUser(request.getName(), request.getEmail(), "secret", Role.ASSEMBLER);
```

Comment:
Every password will be "secret" in the db? Should we store this field in the db, since we have facebook login?

Reply:
oh, yeah, I forgot to change that. I was just thinking what could be the best attempt as we should store some data of the user in the DB, but if the password field is empty, then the form login would match with an empty password field. I was thinking, maybe generating a random UUID and hashing it, till the user changes the password in his profile. So basically we would block the normal login for the facebook users till they change the password